app/controllers/courses_controller.rb

```diff
@@ -54,11 +57,15 @@ def save_course
 
   def destroy_course
     @course = Course.find_by(id: params[:id])
-    if @course.present? && @course.destroy
-      flash[:notice] = "This course is deleted successfully"
+    return invalid_params unless  @course.present?
+    authorize @course if @course.present?
+    instructor = Enrollment.find_by(course_id:  @course.id, enrollment_type: 'instructor')
```

Comment:
```suggestion
    instructor = Enrollment.find_by(course_id: @course.id, enrollment_type: 'instructor')
```

Reply:
I resolved all of the issues related to this issue that I found.
Thanks you